lib/components/user/monitored-trip/trip-status.js

```diff
@@ -0,0 +1,391 @@
+        }
+      }
+
+      body = `Trip is due to begin at ${tripStartMoment.format(timeFormat)}.`
+    } else if (journeyState.tripStatus === 'TRIP_ACTIVE') {
+      // analyze whether the journey state indicates that the matching itinerary
+      // has realtime data. It is assumed that realtime data will exist for
+      // itineraries that have differing values of the scheduled arrival time
+      // and matching itinerary arrival time.
+      if (journeyState.hasRealtimeData) {
+        // calculate the deviation from the scheduled arrival time (positive
+        // value indicates delay)
+        const arrivalDeviationSeconds = (
+          matchingItinerary.endTime -
+          journeyState.scheduledArrivalTimeEpochMillis
+        ) / 1000
+        const deviationHumanDuration = formatDuration(
+          Math.abs(arrivalDeviationSeconds)
+        )
+        if (Math.abs(arrivalDeviationSeconds) < ON_TIME_THRESHOLD_SECONDS) {
+          // about on time
+          headerText = 'Trip is in progress and is about on time.'
+          panelBsStyle = 'success'
+        } else if (arrivalDeviationSeconds > 0) {
+          // delayed
+          headerText = `Trip is in progress and is delayed ${deviationHumanDuration}!`
+          panelBsStyle = 'warning'
+        } else {
+          // early
+          headerText = `Trip is in progress and is arriving ${deviationHumanDuration} earlier than expected!`
+          panelBsStyle = 'warning'
+        }
+      } else {
+        headerText = 'Trip is in progress (no realtime updates available).'
+        panelBsStyle = undefined
+      }
+
+      body = `Trip is due to arrive at the desitination at ${tripEndMoment.format(timeFormat)}.`
+    }
+
+    return {
+      body,
+      headerText,
+      panelBsStyle
+    }
+  }
+
+  renderHeader = () => {
+    const { monitoredTrip } = this.props
+    const journeyState = this.getJourneyState()
+
+    let headerText
+    if (!monitoredTrip.isActive) {
+      headerText = 'Trip monitoring is paused'
+    } else if (monitoredTrip.snoozed) {
+      headerText = 'Trip monitoring is snoozed for today'
+    } else if (journeyState.tripStatus === 'NO_LONGER_POSSIBLE') {
+      headerText = 'Trip is no longer possible.'
+    } else if (journeyState.tripStatus === 'NEXT_TRIP_NOT_POSSIBLE') {
+      headerText = 'Trip is not possible today.'
+    } else if (
+      journeyState.tripStatus === 'TRIP_UPCOMING' ||
+        journeyState.tripStatus === 'TRIP_ACTIVE'
+    ) {
+      headerText = this.renderRealtimeInfo().headerText
+    } else {
+      // trip status not yet calculated, return null
+      return null
+    }
+
+    const secondsSinceLastCheck = moment().diff(
+      moment(journeyState.lastCheckedEpochMillis),
+      'seconds'
+    )
+    const lastCheckedText = `Last checked: ${formatDuration(secondsSinceLastCheck)} ago`
+
+    return (
+      <Panel.Heading>
+        <h3>{headerText}</h3>
+        <h6>{lastCheckedText}</h6>
```

Comment:
Since these are standard HTML tags encapsulated within a bootstrap panel, they don't lack styling rules completely, so there is some styling that gets applied. Not sure what other custom styles we need, so I'm not going to implement anything around this for now.